src/Facebook/InstantArticles/Elements/InstantArticle.php

```diff
@@ -219,6 +219,64 @@ public function withFooter($footer)
     }
 
     /**
+     * Replace all the children within this InstantArticle
+     *
+     * @param Element[] $children Array of elements replacing the original.
+     *
+     * @return $this
+     */
+    public function withChildren($children)
+    {
+        Type::enforceArrayOf(
+            $children,
+            [
+                Ad::getClassName(),
+                Analytics::getClassName(),
+                AnimatedGIF::getClassName(),
+                Audio::getClassName(),
+                Blockquote::getClassName(),
+                Image::getClassName(),
+                H1::getClassName(),
+                H2::getClassName(),
+                Interactive::getClassName(),
+                ListElement::getClassName(),
+                Map::getClassName(),
+                Paragraph::getClassName(),
+                Pullquote::getClassName(),
+                RelatedArticles::getClassName(),
+                Slideshow::getClassName(),
+                SocialEmbed::getClassName(),
+                Video::getClassName()
+            ]
+        );
+        $this->chidren = $children;
+
+        return $this;
+    }
+
+    /**
+     * Replace all the children within this InstantArticle
+     *
+     * @param Type::INTEGER $index The index of the element to delete
+     *                             in the array of children.
+     *
+     * @return $this
+     */
+    public function deleteChild($children)
```

Comment:
Im not sure if this should be $index or if your logic should be something else.
Can you double check?